library/std/src/panicking.rs

```diff
@@ -296,10 +296,17 @@ fn default_hook(info: &PanicInfo<'_>) {
             }
             Some(BacktraceStyle::Off) => {
                 if FIRST_PANIC.swap(false, Ordering::SeqCst) {
-                    let _ = writeln!(
-                        err,
-                        "note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace"
-                    );
+                    let _ = if let Ok(_) = crate::env::var("PSModulePath") {
```

Comment:
Couldn't you just check the $SHELL env variable?

Reply:
Does not exist on Powershell by default, just confirmed on Linux and Windows.